lib/app/features/feed/domain/usecases/feed_toggle_feature.dart

```diff
@@ -10,8 +10,12 @@ class FeedToggleFeature {
   Future<bool> get isEnabled => _isEnabled();
 
   Future<bool> _isEnabled() async {
-    final module =
-        await _modulesServices.feature(name: FeedToggleFeature.featureCode);
-    return module != null;
+    try {
+      final module =
+          await _modulesServices.feature(name: FeedToggleFeature.featureCode);
+      return module != null;
+    } catch (e) {
+      return false;
+    }
```

Comment:
Tratando erro que com origem do módulo de serviço. Na verdade o module não deveria gerar crash já que está sendo consumido por um use case, ele deveria retornar um Either justamente para permitir um tratamento correto

Reply:
ainda tenho minhas duvidas sobre a vantagem de se utilizar `Either`, já que o próprio `Future` provê o `catchError`. :thinking:

Reply:
> ainda tenho minhas duvidas sobre a vantagem de se utilizar `Either`, já que o próprio `Future` provê o `catchError`. 🤔

Não sei se entendi a tua dúvida, porque ao meu olhar são coisas distintas, onde uma não anula a outra, elas se complementam. 

O `Future` é a marcação que o código é assíncrono do código, sendo que no caminho feliz é esperado um valor (ou um void), mas pode ocorrer `Exception` e por isto necessário realizar o tratamento de erro através de um `try { } catch { } `. Em algumas linguagens é possível forçar, pelo compilador, o tratamento do erro, mas não é o caso do Dart, desta maneira o usuário vai receber um crash se o desenvolvedor não for cuidadoso por vontade própria.

Já o `Either` implementa o [**Result pattern**](https://swlaschin.gitbooks.io/fsharpforfunandprofit/content/posts/recipe-part2.html) neste ambiente, deixando bem claro para o desenvolvedor (e para o compilador) que existe duas possibilidades de valores no retorno, um erro ou o resultado.

É possível utilizar o `Either` independente do `Future` e vice-versa. Mas normalmente trabalhamos com a situação de que nos data Sources são a maior fonte de `Exception` e neste momento tratamos no repository (quem normalmente está consumindo o data Source) para converter o resultado em `Either`.

Reply:
> ainda tenho minhas duvidas sobre a vantagem de se utilizar `Either`, já que o próprio `Future` provê o `catchError`. 🤔

É preciso tomar muito cuidado com o `catchError`. Tratamento de erro em nested `Future` é muito complexo e muito sucessível a bug bem dífcil de tratar.

A documentação do Dart que apresenta o `catchError` [Futures and error handling](https://dart.dev/guides/libraries/futures-error-handling) recomenda prefêncialmente utilizar a documentação [handling-errors](https://dart.dev/codelabs/async-await#handling-errors) com async e `try { } catch { }`.

Mas se desja utilizar o `cathError` é muito importante ler [Errors in the middle of a long chain](https://dart.dev/guides/libraries/futures-error-handling#errors-in-the-middle-of-a-long-chain).